### ISSUE TITLE
`jinad` only terminates one of the `shards`

### ISSUE THREAD

**Issue by JoanFM** [jina-ai/jina#1792]
**Describe the bug**
When terminating a remote `Flow` with remote `Peas` it seems that the terminate signal is only received by one of the `shards`

```
      encoder/1@181[I]:recv IndexRequest  from gateway▸encoder/head/ZEDRuntime▸encoder/1/ZEDRuntime▸⚐
      encoder/2@193[I]:#sent: 25 #recv: 13 sent_size: 21.0 MB recv_size: 22.6 MB
      encoder/1@181[I]:#sent: 23 #recv: 12 sent_size: 19.3 MB recv_size: 20.8 MB
   encoder/head@167[I]:recv ControlRequest  from encoder/2▸encoder/head/ZEDRuntime▸⚐
   encoder/tail@174[I]:recv IndexRequest  from gateway▸encoder/head/ZEDRuntime▸encoder/2/ZEDRuntime▸encoder/tail/ZEDRuntime▸⚐
   encoder/tail@174[I]:#sent: 23 #recv: 24 sent_size: 40.3 MB recv_size: 42.1 MB
   encoder/head@167[I]:recv ControlRequest  from encoder/1▸encoder/head/ZEDRuntime▸⚐
   encoder/tail@174[I]:recv IndexRequest  from gateway▸encoder/head/ZEDRuntime▸encoder/1/ZEDRuntime▸encoder/tail/ZEDRuntime▸⚐
   encoder/tail@174[I]:#sent: 24 #recv: 25 sent_size: 42.1 MB recv_size: 43.8 MB
      encoder/2@193[I]:recv ControlRequest  from ctl▸encoder/2/ZEDRuntime▸⚐
      encoder/2@193[I]:#sent: 28 #recv: 14 sent_size: 22.8 MB recv_size: 22.6 MB
      encoder/2@193[I]:no update since 2021-01-26 18:49:29, will not save. If you really want to save it, call "touch()" before "save()" to force saving
      encoder/2@ 1[S]:terminated
👻       PeaStore@ 1[S]:d58ccb61-16ed-4149-8574-5c709e21c0a4 is released from the store.
```

**Comment by JoanFM** [comment]
I think it may be due to having the same `id`:

```
👻       PeaStore@ 1[S]:d58ccb61-16ed-4149-8574-5c709e21c0a4 is added to workspace bc02243c-cb61-48e8-ac9c-e06a063cf139
👻         DAEMON@ 1[S]:/tmp/jinad/bc02243c-cb61-48e8-ac9c-e06a063cf139/d58ccb61-16ed-4149-8574-5c709e21c0a4/logging.log is ready for streaming
👻         DAEMON@ 1[S]:/tmp/jinad/bc02243c-cb61-48e8-ac9c-e06a063cf139/d58ccb61-16ed-4149-8574-5c709e21c0a4/logging.log is ready for streaming
👻         DAEMON@ 1[I]:172.25.32.2:39150 is connected to stream logs!
👻         DAEMON@ 1[S]:/tmp/jinad/bc02243c-cb61-48e8-ac9c-e06a063cf139/d58ccb61-16ed-4149-8574-5c709e21c0a4/logging.log is ready for streaming
👻 WorkspaceStore@ 1[W]:file encoder.yml already exists in workspace bc02243c-cb61-48e8-ac9c-e06a063cf139, will be replaced
👻 WorkspaceStore@ 1[I]:saved uploads to /tmp/jinad/bc02243c-cb61-48e8-ac9c-e06a063cf139/encoder.yml
      encoder/2@193[I]:starting jina.peapods.runtimes.zmq.zed.ZEDRuntime...
      encoder/2@193[I]:input tcp://0.0.0.0:45233 (DEALER_CONNECT) output tcp://0.0.0.0:43833 (PUSH_CONNECT) control over tcp://0.0.0.0:39461 (PAIR_BIND)
ImageTorchEncoder@193[I]:post_init may take some time...
ImageTorchEncoder@193[I]:post_init may take some time takes 0 seconds (0.77s)
ImageTorchEncoder@193[S]:successfully built ImageTorchEncoder from a yaml config
      encoder/2@ 1[S]:ready and listening
   encoder/head@167[I]:recv ControlRequest  from encoder/2▸encoder/head/ZEDRuntime▸⚐
👻       PeaStore@ 1[S]:d58ccb61-16ed-4149-8574-5c709e21c0a4 is added to workspace bc02243c-cb61-48e8-ac9c-e06a063cf139

```

**Comment by hanxiao** [comment]
I believe this is related to the recent PR #1768 on `/flows` endpoint, @deepankarm please add tests to cover this endpoint

**Comment by JoanFM** [comment]
Closed and added warning in #1804
